iup/__init__.py

```diff
@@ -18,15 +18,15 @@ def __init__(self, *args, **kwargs):
         """
         super().__init__(*args, **kwargs)
         self.validate()
+        self.sort("date")
 
     def validate(self):
         """
-        Validate that an UptakeData object has the three key columns:
-        date, grouping factors, and uptake estimate (% of population).
+        Validate that an UptakeData object has the two key columns:
+        date and uptake estimate (% of population). There may be others.
         """
```

Comment:
Since there are multiple functions on `IncidentUptakeData` that require the argument of `group_cols`, maybe it will be helpful to validate the `group_cols` too? As their column names change, maybe we can check that the data should be more than 2 columns (besides `date` and `estimate`)? Just a thought!

Reply:
Good catch! We will likely want to accommodate scenarios with no grouping factors, so I will update the code to recognize "None" as a valid entry for group_cols.